### ISSUE TITLE
 ✨ v6.0.0 ✨

### ISSUE THREAD

**Issue by Salakar** [invertase/react-native-firebase#2025]
Hey everybody 💛 

Super excited to announce that we're just over halfway through our v5 -> v6 migration + overhaul, and; there are so many great things there already for you to try out. We say this every time but really; this is our 'largest release yet' x💯 

So please 🙏 check out the [changelog](https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md) and if all the Firebase services that you use are already listed there then you are good to migrate to the new individual npm packages. They are only released as patch versions for now until all packages are ready - at which point they'll be published as v6.0.0 - I'd recommend locking your package versions until this time.

Instructions can be found on the changelog. Please bare with us well we work towards v6, you have not been forgotten.

I will update this issue with currently being worked on modules and modules next in line, so subscribe to be notified of that and other announcements.

### Blog posts about v6.0.0

 - [Firebase Crashlytics for React Native](https://invertase.io/blog/firebase-crashlytics-for-react-native?utm_source=github&utm_medium=issue2025)
 - [React Native Firebase in 2019](https://invertase.io/blog/react-native-firebase-2019?utm_source=github&utm_medium=issue2025)

---

React Native Firebase is built and maintained with 💛 by [Invertase](https://invertase.io).

- [💼 Hire Us](https://invertase.io?utm_source=github&utm_medium=issue2025)
- [☕️ Sponsor Us](https://opencollective.com/react-native-firebase)

---

**Comment by Salakar** [comment]
----

**In-progress:** Storage

- [WIP Changelog](https://github.com/invertase/react-native-firebase/blob/%40salakar/v6/storage/CHANGELOG.md#storage) ✨

---

**Next-up:** MLKit

---

**Comment by Salakar** [comment]
[reserved]

**Comment by Salakar** [comment]
[reserved]

**Comment by paolospag** [comment]
This is a good news @Salakar !
I have two questions:

1) why are [Authentication ](https://www.npmjs.com/package/@react-native-firebase/auth) and [Cloud Firestore](https://www.npmjs.com/package/@react-native-firebase/firestore) services missing in the changelog despite being available in npm?
2) what is the release estimation for the Cloud Messaging service?

**Comment by Salakar** [comment]
@paolospag hey, sure;

1. they're just placeholder packages, changelog is the source of truth on what is available
2. it's last in line just because of it's size and didn't want to hold up the others too much, can't give an ETA on that yet because of that and that I need to scope out how far to go with it change and feature wise - might make an issue to discuss that in a few days

**Comment by paolospag** [comment]
Thanks for clarifying.
So, what is the ETA for Authentication and Cloud Firestore?

**Comment by Salakar** [comment]
> So, what is the ETA for Authentication and Cloud Firestore?

These are being worked alongside the other modules in a separate change list. Can't give you an ETA right now but they're definitely in progress 🙃

**Comment by loga4** [comment]
Hi! It's a great work! Waiting for a RC :)

I would like to know when it is planned to migration remote push notification? Because in the v5.x.x version of the push come, but listen handlers do not work, except getInitialNotification

**Comment by Salakar** [comment]
@loga4 it's last on the list just because it needs the most attention and I didn't want to hold up all the other modules because of it

**Comment by salmazov** [comment]
@Salakar I want to use RNFirebase with react-native 0.59. What is the version of RNFirebase I should use?

**Comment by mikehardy** [comment]
@salmazov I believe according to the v6 documentation, if you only need the modules already ported, you should use v6 if possible. If like me you need unported modules (auth, firestore, etc) then v5.3.1 is it. Be aware that on the v5.3.1 you need to be current - 0.59.3+ and gradle version and gradle deps and Firebase Pods up to date etc - the versions are very important

**Comment by salmazov** [comment]
I have 
"react-native": "0.59.4",
"react-native-firebase": "^5.3.1",

And Podfile
  pod 'Firebase/Core'
  pod 'RNFirebase', :path => '../node_modules/react-native-firebase/ios'

  pod 'Firebase/AdMob'
  pod 'Firebase/Analytics'
  pod 'Firebase/Auth'
  pod 'Firebase/Crash'
  pod 'Firebase/Database'
  pod 'Firebase/DynamicLinks'
  pod 'Firebase/Messaging'
  pod 'Firebase/RemoteConfig'
  pod 'Firebase/Storage'

My app crashes with
Thread 1: signal SIGABRT

`Require cycles are allowed, but can result in uninitialized values. Consider refactoring to remove the need for a cycle.
2019-04-16 09:52:09.723321+0300 NaturnikiMobile[3767:27165] 5.9.0 - [Firebase/Core][I-COR000004] App with name __FIRAPP_DEFAULT does not exist.
2019-04-16 09:52:09.731154+0300 NaturnikiMobile[3767:26898] *** Terminating app due to uncaught exception 'com.firebase.core', reason: 'Name cannot be __FIRAPP_DEFAULT.'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000115b821bb __exceptionPreprocess + 331
	1   libobjc.A.dylib                     0x000000011511c735 objc_exception_throw + 48
	2   CoreFoundation                      0x0000000115b82015 +[NSException raise:format:] + 197
	3   NaturnikiMobile                     0x000000010df16cb8 +[FIRApp configureWithName:options:] + 312
	4   NaturnikiMobile                     0x000000010e1a9e33 __45-[RNFirebase initializeApp:options:callback:]_block_invoke + 1187
	5   NaturnikiMobile                     0x000000010e2c0f8d __RCTUnsafeExecuteOnMainQueueSync_block_invoke + 29
	6   libdispatch.dylib                   0x0000000117587602 _dispatch_client_callout + 8
	7   libdispatch.dylib                   0x00000001175957be _dispatch_async_and_wait_invoke + 176
	8   libdispatch.dylib                   0x0000000117587602 _dispatch_client_callout + 8
	9   libdispatch.dylib                   0x000000011759499a _dispatch_main_queue_callback_4CF + 1541
	10  CoreFoundation                      0x0000000115ae73e9 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	11  CoreFoundation                      0x0000000115ae1a76 __CFRunLoopRun + 2342
	12  CoreFoundation                      0x0000000115ae0e11 CFRunLoopRunSpecific + 625
	13  GraphicsServices                    0x00000001186181dd GSEventRunModal + 62
	14  UIKitCore                           0x000000011ef1a81d UIApplicationMain + 140
	15  NaturnikiMobile                     0x000000010deb2700 main + 112
	16  libdyld.dylib                       0x00000001175fd575 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) `

**Comment by mikehardy** [comment]
@salmazov you have posted about a new issue regarding 5.3.x in the 6.x thread. I don't think you'll find answers here, especially since so much information is missing - open a new issue following the template so all necessary information is included

**Comment by salmazov** [comment]
> @salmazov you have posted about a new issue regarding 5.3.x in the 6.x thread. I don't think you'll find answers here, especially since so much information is missing - open a new issue following the template so all necessary information is included

Ok!

But I have different question. How to install v.6 instead 5.3.3 via npm?

**Comment by mikehardy** [comment]
@salmazov by reading the docs in the respective modules, e.g. https://github.com/invertase/react-native-firebase/tree/master/packages/analytics - the level of resourcefulness required to get v6 working may very well be higher than the amount needed to find that link so you may find the experience frustrating vs the  5.3 series.

**Comment by salmazov** [comment]
> @salmazov by reading the docs in the respective modules, e.g. https://github.com/invertase/react-native-firebase/tree/master/packages/analytics - the level of resourcefulness required to get v6 working may very well be higher than the amount needed to find that link so you may find the experience frustrating vs the 5.3 series.

Does it mean I should install modules separately? I only need to know how install 6 via npm )

**Comment by salmazov** [comment]
Where I can find packages for Storage/Database?

**Comment by mikehardy** [comment]
@salmazov the level of resourcefulness required to get v6 working may very well be higher than the amount needed to find that link so you may find the experience frustrating vs the 5.3 series.

You need to become very comfortable with reading the docs and trying things.

**Comment by salmazov** [comment]
> @salmazov the level of resourcefulness required to get v6 working may very well be higher than the amount needed to find that link so you may find the experience frustrating vs the 5.3 series.
> 
> You need to become very comfortable with reading the docs and trying things.

Thanks ) I found information

**Comment by mikehardy** [comment]
> > @salmazov t

> Thanks ) I found information

:muscle: :smile:

**Comment by Salakar** [comment]
@mikehardy looks like they deleted their comments ;D

**Comment by mazenchami** [comment]
@Salakar when will `v6` be pushed?
I am currently looking to migrate away from Google Analytics and trying to decide if I should just wait for `v6` as the latest version installation seems to be _MUCH_ longer than `v6`...

**Comment by Salakar** [comment]
@mazenchami if you just need Firebase Analytics then v6 is ready to go for you, see the change log for more info.

**Comment by addu1** [comment]
Are these methods for storage available now?
// Pause the upload
uploadTask.pause();

// Resume the upload
uploadTask.resume();

// Cancel the upload
uploadTask.cancel()

**Comment by Salakar** [comment]
@addu1 yes they are 🎉 PR still in progress though here: https://github.com/invertase/react-native-firebase/pull/2043 - but check the changelog for storage: https://github.com/invertase/react-native-firebase/blob/%40salakar/v6/storage/CHANGELOG.md#storage - many new things.

**Comment by mazenchami** [comment]
> @mazenchami if you just need Firebase Analytics then v6 is ready to go for you, see the change log for more info.

@Salakar thank you very much for the info. Is Crashlytics also ready in v6? Additionally, if I wanted to manually install the package, if I follow the [RN iOS Manual](https://facebook.github.io/react-native/docs/linking-libraries-ios#manual-linking) and the usual Android files `MainApplication, settings.gradle, & /app/build.gradle`?

**Comment by Salakar** [comment]
@mazenchami yes it is. You can install using RN link as well:

```bash
yarn add @react-native-firebase/app@alpha
yarn add @react-native-firebase/crashlytics@alpha

react-native link @react-native-firebase/app
react-native link @react-native-firebase/crashlytics
```

(app is required by all modules)

**Comment by mazenchami** [comment]
@Salakar thanks for the detail. don't mean to be a pest, please let me know if there is another place to post my questions. My guess is that the only other item needed is the downloaded `GoogleService-Info.plist` file?

**Comment by Salakar** [comment]
Yes, you'll need the Google Services plist/json for iOS/Android as well sorry.

If you have any more questions feel free to join the v6 discord server mentioned in the 4th comment above - will be easier :)

**Comment by Salakar** [comment]
Hey everyone 👋 

I think we're in a comfortable place to open up the new site. 

**Preview URL:** https://dev.invertase.io/oss/react-native-firebase

There's still on-going work to improve it and a lot of content missing (or needs re-working); however, we're making it public so that we can get your feedback and also open it up to anyone wanting to submit content (PRs 💯)

### Submitting Feedback

If you have any feedback please either submit it here, on the RNFB v6 discord or using the `Send Feedback` button that appears on most pages:

![image](https://user-images.githubusercontent.com/5347038/56816478-8679d000-683b-11e9-87c5-33bea121b629.png)

The more detail you provide the better.

### Contributing

If you'd like to contribute to the documentation here's a quick overview of how it works.

 - **General Documentation**: All pages have an `Edit page` button at the top that will quickly take you to the relevant markdown file that generates the content of the page - feel free to fork the repo and edit these and send up a PR.
![image](https://user-images.githubusercontent.com/5347038/56816478-8679d000-683b-11e9-87c5-33bea121b629.png)
 - **Reference documentation**: All Reference documentation is generated from the `index.d.ts` type definition file for each module e.g. `packages/analytics/index.d.ts`, if you'd like to add examples, parameter descriptions, method descriptions for anything that is lacking them, please also send a PR that adds code comments to the `index.d.ts` files (many of these are missing annotations and need going through).

If you have any questions or queries on how to contribute please get in touch, we'd be happy to help or mentor if required.

**Comment by Salakar** [comment]
**Cloud Storage** is now up, published in alpha10. 

[[View the Cloud Storage changelog]](https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md#storage)

**Comment by Salakar** [comment]
**Authentication** in progress: https://github.com/invertase/react-native-firebase/pull/2107

**Dynamic Links** in progress: https://github.com/invertase/react-native-firebase/pull/2103

**Comment by Salakar** [comment]
**Dynamic Links** in progress: https://github.com/invertase/react-native-firebase/pull/2103

**Comment by Salakar** [comment]
**ML-Kit** in progress: https://github.com/invertase/react-native-firebase/pull/2117

**Comment by Salakar** [comment]
👋 `6.0.0-alpha.11` published to NPM - now with Authentication (TS types support only - flow and other things in another alpha, see https://github.com/invertase/react-native-firebase/pull/2113).

**Comment by hellpirat** [comment]
Hi, Is this ready for using firebase/Messaging? I would like to try it :)  I have a use_frameworks in my pod, so will it work for me? 🤔

**Comment by salmazov** [comment]
I want to use only Firebase Auth/Storage. What is the Pod configuration I need when I use react-native-firebase v.6?

**Comment by Salakar** [comment]
@salmazov I'd recommend following through the new projects guide: https://invertase.io/oss/react-native-firebase/quick-start/new-project and then these guides:

 - auth: https://invertase.io/oss/react-native-firebase/v6/auth/quick-start
 - storage: https://invertase.io/oss/react-native-firebase/v6/storage/quick-start

**Comment by ouabing** [comment]
Hey guys, I'm managed to migrate my old Fabric Crashlytics to Firebase, I just wonder why the Crashlytics Podspec targets iOS 10.0? I know App Store may will reject apps targeting 9.0 later, but there are still 1% iOS 9.0 users using our app.

**Comment by Salakar** [comment]
@ouabing we'll probably lower it down in another alpha, the decision was mainly for Notifications - but we can just make that one specifically iOS10

**Comment by ouabing** [comment]
@Salakar Got it, thanks for the quick reply!

**Comment by HuuNguyen312** [comment]
Hi guys, I can't add firebase message to my project.

`yarn add @react-native-firebase/messaging@alpha
yarn add v1.9.4
warning ../../../../package.json: No license field
[1/4] 🔍  Resolving packages...
error An unexpected error occurred: "https://registry.yarnpkg.com/@react-native-firebase%2fmessaging: Not found".
info If you think this is a bug, please open a bug report with the information provided in "/Users/huunn/WorkSpace/React-Native/Asio/Upde/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
`
Please help me resolve it.
Thanks

**Comment by mikehardy** [comment]
@HuuNguyen312 open a new issue, follow the issue template carefully

**Comment by Salakar** [comment]
@HuuNguyen312 , and also, messaging doesn't exist yet as mentioned at the top of this issue

**Comment by sibelius** [comment]
`npx @react-native-community/cli@next init --template=@react-native-firebase/template@alpha Awesome`

is throwing the following error

```jsx
✔ Downloading template
✔ Copying template
✔ Processing template
⠋ Installing all required dependenciesUsing React Native Firebase version '6.0.0-alpha.22'
 -> Detecting Firebase modules...
    ✓ Discovered 'app' (+ build script)
Analyzing dependencies
Fetching podspec for `DoubleConversion` from `../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec`
Fetching podspec for `Folly` from `../node_modules/react-native/third-party-podspecs/Folly.podspec`
Fetching podspec for `RNFBApp` from `../node_modules/@react-native-firebase/app`
Fetching podspec for `React` from `../node_modules/react-native`
Fetching podspec for `glog` from `../node_modules/react-native/third-party-podspecs/glog.podspec`
Fetching podspec for `yoga` from `../node_modules/react-native/ReactCommon/yoga`
[!] CocoaPods could not find compatible versions for pod "Firebase/Core":
  In Podfile:
    RNFBApp (from `../node_modules/@react-native-firebase/app`) was resolved to 6.0.0-alpha.22, which depends on
      Firebase/Core (~> 5.20.2)

None of your spec sources contain a spec satisfying the dependency: `Firebase/Core (~> 5.20.2)`.

You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.

Note: as of CocoaPods 1.0, `pod repo update` does not happen on `pod install` by default.

✖ Installing all required dependencies
error Error: Failed to install CocoaPods dependencies for iOS project, which is required by this template.
Please try again manually: "cd ./Awesome/ios && pod install".
CocoaPods documentation: https://cocoapods.org/
/Users/sibelius/.npm/_npx/70520/lib/node_modules/@react-native-community/cli/node_modules/graceful-fs/polyfills.js:10
    cwd = origCwd.call(process)
                  ^

Error: ENOENT: no such file or directory, uv_cwd
    at process.cwd (/Users/sibelius/.npm/_npx/70520/lib/node_modules/@react-native-community/cli/node_modules/graceful-fs/polyfills.js:10:19)
    at setopts (/Users/sibelius/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/glob/common.js:93:21)
    at new Glob (/Users/sibelius/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/glob/glob.js:135:3)
    at Function.glob.hasMagic (/Users/sibelius/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/glob/glob.js:101:11)
    at Function.rimrafSync [as sync] (/Users/sibelius/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/rimraf/rimraf.js:273:36)
    at process.on (/Users/sibelius/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/libnpx/index.js:151:37)
    at process.emit (events.js:182:13)
    at processEmit [as emit] (/Users/sibelius/.npm/_npx/70520/lib/node_modules/@react-native-community/cli/node_modules/signal-exit/index.js:149:35)
```

**Comment by Salakar** [comment]
@sibelius you might need to run `pod repo update` on your terminal first, I think your specs may be out of date - seems to runs fine for me on latest versions

**Comment by sibelius** [comment]
I need to run `pod repo update` before creating a new project?

**Comment by Salakar** [comment]
@sibelius just once for this time should be fine afterwards.

cc @thymikee - I think the CLI should probably be running `pod install --repo-update` instead of just install, otherwise this happens, which fails the template creation also, thoughts?

**Comment by thymikee** [comment]
Yep! Send PRs pretty pls

**Comment by Salakar** [comment]
> Yep! Send PRs pretty pls

![maybe gif](https://media.giphy.com/media/sAO5iUp7tv6b6/giphy.gif)

🤣 but sure, won't be able to until next week, busy at the moment - which approach do you think is best, `--repo-update` by default may be a bit intense given that RN only uses local Pods.

**Comment by thymikee** [comment]
Let's find out! cc @orta @alloy

**Comment by sibelius** [comment]
not sure if this is a problem with react-native-firebase template or cli, but most files are executables

![image](https://user-images.githubusercontent.com/2005841/58711793-908c7280-8395-11e9-9868-80e973a99c13.png)

I don''t think App.js should be executable

**Comment by orta** [comment]
Yep, as the firebase pods you're depending on comes from the main CocoaPods trunk specs repo - then you'll need to update your local specs in order to resolve a copy of them

alternative is to bundle the pod specs but that's gonna be a lot of maintaining

**Comment by mikehardy** [comment]
In general would you agree: the only way to be sure, during a pod link, is to update the specs first before any new link? (I would assert that statement is true but my XCode dev experience is still so shallow)

**Comment by orta** [comment]
Nope, most RN libs wouldn't need it - only libs with dependencies outside of the node_modules would need that.

**Comment by mikehardy** [comment]
Oh - that makes perfect sense. Thanks for the reply! I suppose there is some CLI magic that can figure it out and only update if it makes sense but my CLI-fu is also weak so I'll listen and see.

**Comment by sibelius** [comment]
I'm checking the docs here:

https://github.com/invertase/react-native-firebase/tree/master/packages/app

does `react-native link` still makes sense? as we are moving to `autolinking`?

**Comment by sibelius** [comment]
I've got another error when using platform: 9.0

```jsx
[!] CocoaPods could not find compatible versions for pod "RNFBApp":
  In Podfile:
    RNFBApp (from `../node_modules/@react-native-firebase/app`)

Specs satisfying the `RNFBApp (from `../node_modules/@react-native-firebase/app`)` dependency were found, but they required a higher minimum deployment target.
```

is iOS 10.0 the mininum version on react-native-firebase 6.x?

**Comment by sibelius** [comment]
Based on this https://github.com/invertase/react-native-firebase/pull/1864

can we mix v6 with v5 until all packages are migrated to v6?

is there a status for all the packages?

**Comment by mikehardy** [comment]
@sibelius ios10 is the minimum, apparently the notifications changes in iOS 10 (vs 9) were pretty large, and carrying them in to the next major version here is painful so they're out for now - I asked the same earlier. I wouldn't place a bet, and I did ask in other locations, but I'd be surprised if react-native supported iOS 9 that much longer, though retiring it will require removing entire devices from compatibility lists whereas the previous minimum iOS supported change did not remove support for any devices, so who knows

As for mixing, I thought the same but haven't tried it. I will be a little surprised if it works, but who knows.

If it's a strong need, maybe porting the ios 9 notifications to react-native-firebase v6 is easiest...

**Comment by Salakar** [comment]
👋 `6.0.0-alpha.24` published to NPM;

 - 🎉 **Real-time Database** landed; but has some minor known issues that will be fixed in the next alpha:
   - No `.push()` ref support
   - Require cycle console warning

----


 - 🎉 **ML Kit Natural Language** landed; supports the following APIs:
   - Language Identification
   - Smart Replies

Examples [in Changelog](https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md#ml-kit-natural-language-mlkitlanguage---new). 

----


 - 🌝 **Remote Config** reworked (again, sorry). All Remote Config values can now be accessed synchronously in JavaScript, see the full description in the [updated changelog](https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md#remote-config-config).

----


 - ⬇️ **iOS** Lowered the deployment target minimum to iOS 9
    - cc @sibelius, @mikehardy - lowered as per the above conversation. Notifications will probably be the only package with an iOS 10 minimum deployment target - so we shouldn't penalise the rest of the packages. We can look at community contribs to bring < iOS 10 support to Notifications if we're unable to ourselves.

----


 - [[Changelog](https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md)]
 - Documentation is not yet deployed
   - cc @ehesp - Reference docs not showing for DB & ML (TS Docs have been generated, reference sections don't show on the sidebars or in the build logs)

**Comment by Ehesp** [comment]
Hi all - `6.0.0-alpha.25` published to NPM.

This fixes the known Database issues above:
- `.push()` support is now in place.
- Require cycle console warning removed.

**Comment by Salakar** [comment]
Hey all; `0.0.26` is up.

----

- 🔗 **Dynamic Links:** 
    - [[changelog]](https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md#dynamic-links-links)

----

- 📇 **App Indexing:** 
    - [[changelog]](https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md#app-indexing-indexing---new)

----

**Small note**: I've switched to publishing all the new packages to the `latest` tag now on NPM but only as patch versions still, won't be v6.0.0 until they're all ready. 

I've done this as publishing a new package to NPM even if tagged will still create a `latest` release - and some people end up using the `latest` ones (which in most cases are just a placeholder) instead of the tagged `alpha` ones which are up to date. 

So as of this message, the latest version you want to be installing is `0.0.26`, I'd recommend also locking your package versions so as not to get surprise patch updates when npm installing.

**New packages**:

```
 - @react-native-firebase/app-types@0.0.26
 - @react-native-firebase/app@0.0.26
 - @react-native-firebase/auth@0.0.26
 - @react-native-firebase/common@0.0.26
 - @react-native-firebase/config@0.0.26
 - @react-native-firebase/crashlytics@0.0.26
 - @react-native-firebase/database@0.0.26
 - @react-native-firebase/fiam@0.0.26
 - @react-native-firebase/firestore@0.0.26
 - @react-native-firebase/iid@0.0.26
 - @react-native-firebase/indexing@0.0.26
 - @react-native-firebase/invites@0.0.26
 - @react-native-firebase/links@0.0.26
 - @react-native-firebase/ml-natural-language@0.0.26
 - @react-native-firebase/perf@0.0.26
 - @react-native-firebase/storage@0.0.26
 - @react-native-firebase/template@0.0.26
 - @react-native-firebase/utils@0.0.26
```

**Comment by Salakar** [comment]
👋 	the following modules are now in progress and quite far along:

 - Cloud Firestore (`firestore()`)
 - Firebase Cloud Messaging (`messaging()`) (messaging only, no notifications yet)

----

As for Notifications `notifications()` - I'd like to start some discussion on this and the feature sets that should land in v6 and future releases. If you're using notifications heavily and have the time to join in the discussions please do so on the `#notifications` channel on this Discord server: https://discord.gg/rHjpxts

**Comment by Salakar** [comment]
v6 Firestore PR is now up: https://github.com/invertase/react-native-firebase/pull/2319 by @Ehesp.

Only a few small things left to do on it, specifically types & transactions.

If you're curious on what's changed, check out the changelog for it: [[changelog]](https://github.com/invertase/react-native-firebase/blob/@ehesp/v6/firestore/CHANGELOG.md#cloud-firestore-firestore)

**Comment by cjadhav** [comment]
Hi Guys,

Thanks for comments it helped to solve the issues to implement V6 in my application.
The main purpose of using firebase is for analytics only.
I am able to implement V6 with new RN 0.60 on iOS with cocoa pods. but facing weird problem for android.
The error is coming as soon as app opens, (no compile time errors)
`RNFBAnalyticsModule tried to override ReactNativeFirebaseAnalyticsModule`

I have updated my system to latest and configuration as as follows,
```
System:
    OS: macOS 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Memory: 824.24 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.15.3 - /usr/local/bin/node
    Yarn: 1.16.0 - /usr/local/bin/yarn
    npm: 6.4.1 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
  IDEs:
    Android Studio: 3.4 AI-183.6156.11.34.5522156
    Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.8.6 => 16.8.6 
    react-native: 0.60.0 => 0.60.0 
  npmGlobalPackages:
    react-native-cli: 2.0.1
    react-native-create-library: 3.1.2
    react-native-git-upgrade: 0.2.7
```
And package.json is 
```
"dependencies": {
    "@react-native-community/async-storage": "^1.5.0",
    "@react-native-community/eslint-config": "^0.0.5",
    "@react-native-community/netinfo": "^4.0.0-alpha.1",
    "@react-native-firebase/analytics": "^0.0.26",
    "@react-native-firebase/app": "^0.0.26",
    "bugsnag-react-native": "2.21.1",
    "dateformat": "3.0.3",
    "eslint": "^6.0.1",
    "i18n-js": "^3.3.0",
    "lodash": "4.17.11",
    "moment": "2.24.0",
    "react": "16.8.6",
    "react-native": "0.60.0",
    "react-native-dates": "2.0.4",
    "react-native-device-info": "2.1.2",
    "react-native-loading-spinner-overlay": "1.0.1",
    "react-native-maps": "^0.24.2",
    "react-native-responsive-ui": "2.1.1",
    "react-native-router-flux": "4.0.6",
    "react-native-scrollable-tab-view2": "0.8.2",
    "react-native-splash-screen": "3.2.0",
    "react-native-statusbar-alert": "0.4.0",
    "react-native-webview": "6.3.1",
    "schedule": "0.5.0"
  },
  "devDependencies": {
    "@babel/core": "^7.4.5",
    "@babel/runtime": "^7.4.5",
    "babel-jest": "24.8.0",
    "jest": "24.8.0",
    "metro-react-native-babel-preset": "0.54.1",
    "react-test-renderer": "16.8.6"
  },
```

In android, (Gradle 5.4.1)
`<project>/build.gradle`
```
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    ext {
        buildToolsVersion = "28.0.3"
        minSdkVersion = 16
        compileSdkVersion = 28
        targetSdkVersion = 28
        supportLibVersion = "28.0.0"
    }
    repositories {
        google()
        jcenter()
        maven {
            url 'https://maven.google.com/'
            name 'Google'
        }
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.4.1'
        classpath 'com.google.gms:google-services:4.2.0'
        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        mavenLocal()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
        maven {
            // Android JSC is installed from npm
            url("$rootDir/../node_modules/jsc-android/dist")
        }
        google()
        jcenter()
        maven {
            url 'https://maven.google.com/'
            name 'Google'
        }
    }
}

subprojects {
    afterEvaluate { project ->
        if (project.hasProperty("android")) {
            android {
                compileSdkVersion rootProject.ext.compileSdkVersion        // version of compile sdk used for project
                buildToolsVersion rootProject.ext.buildToolsVersion    // version of build tool used for project

                lintOptions {
                    abortOnError false
                }
            }
        }
    }
}
```

`<project>/app/build.gradle`
```

dependencies {
    implementation project(':@react-native-firebase_analytics')
//    implementation 'com.google.firebase:firebase-core:17.0.0'
    implementation project(':@react-native-firebase_app')
    implementation project(':@react-native-community_netinfo')
    implementation project(':@react-native-community_async-storage')
    implementation project(':react-native-webview')
    implementation project(':react-native-splash-screen')
    implementation project(':react-native-device-info')
    implementation project(':bugsnag-react-native')
    implementation project(':react-native-maps')
    implementation(project(':react-native-maps')) {
        exclude group: 'com.google.android.gms', module: 'play-services-base'
        exclude group: 'com.google.android.gms', module: 'play-services-maps'
    }
    implementation ('com.google.android.gms:play-services-gcm:15.0.1') {
        force = true;
    }
    implementation 'com.google.android.gms:play-services-base:+'
    implementation 'com.google.android.gms:play-services-maps:+'
    implementation 'com.google.android.gms:play-services-location:+'
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.facebook.react:react-native:+"  // From node_modules

    // JSC from node_modules
    if (useIntlJsc) {
        implementation 'org.webkit:android-jsc-intl:+'
    } else {
        implementation 'org.webkit:android-jsc:+'
    }
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)

apply plugin: 'com.google.gms.google-services'
``` 

also updated `MainApplication.java` as per RN0.60
```
@Override
    protected List<ReactPackage> getPackages() {
        @SuppressWarnings("UnnecessaryLocalVariable")
        List<ReactPackage> packages = new PackageList(this).getPackages();
      // Packages that cannot be autolinked yet can be added manually here, for example:
      // packages.add(new MainReactPackage());
        packages.add(new ReactNativeFirebaseAnalyticsPackage());
        packages.add(new ReactNativeFirebaseAppPackage());
        packages.add(new NetInfoPackage());
        packages.add(new AsyncStoragePackage());
        packages.add(new RNCWebViewPackage());
        packages.add(new SplashScreenReactPackage());
        packages.add(new MapsPackage());
        packages.add(new RNDeviceInfo());
        packages.add(BugsnagReactNative.getPackage());
      return packages;
    }
```

I am not sure is this right place to share the issue. I hoping to get some guidance.

Thanks!

**Comment by Salakar** [comment]
@cjadhav RN 0.60 has auto-linking which RNFB v6 supports - you don't need to add these to your gradle files, I'd remove the following from your project:

```
        packages.add(new ReactNativeFirebaseAnalyticsPackage());
        packages.add(new ReactNativeFirebaseAppPackage());
```

```
    implementation project(':@react-native-firebase_analytics')
//    implementation 'com.google.firebase:firebase-core:17.0.0'
    implementation project(':@react-native-firebase_app')
```

(also remove them from `settings.gradle` if you've added them there too)


You'll probably need to do this for the other modules that you have manually linked as well, afaik WebView, NetInfo, DeviceInfo etc all support auto-linking

**Comment by Salakar** [comment]
Howdy 👋 

The Messaging branch is up, there are still a few things left to do, but, for those interested here's the changelog:

[[Changelog]](https://github.com/invertase/react-native-firebase/blob/%40salakar/v6/messaging/CHANGELOG.md#cloud-messaging-messaging)

👀

**Comment by andrewhartwig** [comment]
Hey guys, I know the documentation is incomplete, and so I seem to be missing a key step somewhere. I've followed all the v6 set up guides, but it seems like something is still an issue. 

```
    "@react-native-firebase/app": "^0.0.26",
    "@react-native-firebase/crashlytics": "^0.0.26",
```

`import crashlytics from '@react-native-firebase/crashlytics';`

```javascript
const App = () => {
  crashlytics.crash();
```

Based on the documentation this should work, but I'm getting two separate errors:

Android: `No Firebase App '[DEFAULT]' has been created - call firebase.initializeApp()`. I tried doing that but I'm missing parameters and don't see any documentation on how to initalize firebase.

iOS: `_crashlytics.default.carsh is not a function`

**Comment by andrewhartwig** [comment]
Will the `v5.x.x` versions of RNFB work with the latest `0.60` RN?

**Comment by mikehardy** [comment]
@andrewhartwig v5xx is supposed to work with RN0.60, but you will need the jetifier packager (which you will likely need for other libraries as well). I'm currently converting my main project to RN60 and although it is painful, I'm not currently blocked by react-native-firebase

**Comment by bakaa99** [comment]
will i have to use jettifier for v6 too?

**Comment by mikehardy** [comment]
@bakaa99 I can't say for certain because I don't know what dependencies you have, but you won't need jetify for *react-native-firebase* v6 (as it is AndroidX already) or RN0.60 (already AndroidX) but I anticipate there will be a large number of libraries that don't update for a while (maintainers on vacation or whatever) so you will likely need to jetify for a little while during the transition. If you're not sure, you can delete node_modules, re-install all your modules, then try it clean or try the `./gradlew :app:dependencies` to check. Or alternatively, if you try react-native-firebase v6 but you are stuck on older react-native (maybe because Expo), then you'll need jetify -r to maintain support library alignment.

**Comment by Salakar** [comment]
Friday surprise 🎉 

The following have landed on master;

 - Cloud Firestore - [[changelog]](https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md#cloud-firestore-firestore)
 - ML Kit Vision - [[changelog]](https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md#ml-kit-vision-mlkitvision---new)
 - Cloud Messaging (not Notifications) - [[changelog]](https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md#cloud-messaging-messaging)

Still a few bits left to do on them before we can publish an alpha though.

Side note; Merging those three above brings us well over 1,000 tests 😱 

![image](https://user-images.githubusercontent.com/5347038/61143021-09203c00-a4c9-11e9-90ac-7da467febd39.png)

---

Think the work we're doing for v6.0.0 is great? Please consider supporting the project with any of the below, thank you.

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

**Comment by paolospag** [comment]
@Salakar amazing news!

Are there any news regarding notifications? What is the ETA?

**Comment by marcomarasco** [comment]
@Salakar good job! how i can test this new release on master? is already available on NPM?

**Comment by congnguyen91** [comment]
Hi, is the messaging package released on master? :) 
I tried to install by yarn but it occurred error 
```
yarn add @react-native-firebase/messaging --force
yarn add v1.17.3
warning ../package.json: No license field
[1/4] 🔍  Resolving packages...
error An unexpected error occurred: "https://registry.yarnpkg.com/@react-native-firebase%2fmessaging: Not found".
info If you think this is a bug, please open a bug report with the information provided in ..
```

**Comment by Psiiirus** [comment]
Can some body pleas tell me which version of the Firebase iOS SDK i have to install if i use version 6 ?

**Comment by mikehardy** [comment]
@Psiiirus I believe it will work tiwh 5.20.x or later but I also believe it works well with "current", so I would install the most current. https://firebase.google.com/support/release-notes/ios says that's 6.5.0 right now

**Comment by Salakar** [comment]
> @Psiiirus I believe it will work tiwh 5.20.x or later but I also believe it works well with "current", so I would install the most current. [firebase.google.com/support/release-notes/ios](https://firebase.google.com/support/release-notes/ios) says that's 6.5.0 right now

Correct, master is also on 6.5.0, but it was on 5.20.x previously.

**Comment by Psiiirus** [comment]
@mikehardy  @Salakar  thanks your reply...I downgraded back to rnf 5.5 with SDK 6.5 which worked fine on android and iOS ...think i'll have to upgrade my iOS to Pods usage to be state of the art and avoid unnecessary errors ^^

**Comment by AilsonFreire** [comment]
I'm facing the same problem as @congnguyen91:
  Resolving packages...
error An unexpected error occurred: "https://registry.yarnpkg.com/@react-native-firebase%2fmessaging: Not found".

**Comment by rgommezz** [comment]
@Salakar what's the approx ETA for an alpha release on Cloud Firestore? 🤖

**Comment by Ehesp** [comment]
@rgommezz We just have iOS transactions left, so should be released by the end of the month!

All of the tests are there & work for Android. If anyone feels up to it, finishing off the iOS code and sending a PR would be awesome.

**Comment by rgommezz** [comment]
Thanks very much for the update @Ehesp, looking forward!

Awesome work guys! 🥇

**Comment by bosung90** [comment]
Is @react-native-firebase/firestore supposed to work for Android? I was getting some compile errors after I did `react-native link  @react-native-firebase/firestore `

**Comment by sudomann** [comment]
>Many of the manual native installation steps for Android & iOS have been removed / internally automated and most modules can now be used just by linking it (e.g. react-native link @react-native-firebase/analytics).

Now that React Native 0.60.x is doing away with linking of native packages, how would one go about using your packages?

**Comment by Ehesp** [comment]
> Is @react-native-firebase/firestore supposed to work for Android? I was getting some compile errors after I did `react-native link @react-native-firebase/firestore `

See message above, it's not released yet.

**Comment by fungilation** [comment]
> > Many of the manual native installation steps for Android & iOS have been removed / internally automated and most modules can now be used just by linking it (e.g. react-native link @react-native-firebase/analytics).
> 
> Now that React Native 0.60.x is doing away with linking of native packages, how would one go about using your packages?

Ya, I can't get 5.5.5 to work on RN 0.60.4 on Android. iOS works. Anyone else have trouble on android? https://github.com/invertase/react-native-firebase/issues/2411

**Comment by dzilva** [comment]
v5.2.1 also have issue. any solution or version for this issue?

**Comment by mikehardy** [comment]
@dzilva - sorry - I can't support anything that old - my main focus is on getting v5 working correctly with the current versions of the rest of the ecosystem (up to date Firebase SDKs, up to date react-native) so I won't offer support for that.

I do have react-native 0.59 working great for Android and iOS on v5.5.5 plus 2 PRs (#2415 #2414) examine https://github.com/mikehardy/rnfbdemo/blob/master/make-demo-rn59.sh

 @fungilation I have react-native 0.60 working fine on Android (doesn't need #2415 - works with v5.5.5 stock) examine https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh

None of this has anything to do with v6.0.0 of react-native-firebase though, so I'm going to minimize all these comments.

**Comment by dzilva** [comment]
I'm getting this error,

RNfirebase core module was not found natively on android

Can you help me to fix? I already tried clean and rebuild, version downgrading. not working..
I'm using Expo + RN

**Comment by mikehardy** [comment]
Expo, not ExpoKit? That will never work with native modules. Fundamental property of Expo. It's in the docs

**Comment by dzilva** [comment]
Much appreciate your comment. I'm a beginner so sorry.
Can I fix it? still I'm getting same error.
I'm running Expo and two physical devices (iPhone and android)
How can I connect to firebase for phone auth?

**Comment by creative-git** [comment]
is database working well with V6 ? 
I followed the documentation , but ios building fails after installing database
RN: 0.60
RNFB: v6

**Comment by Ehesp** [comment]
@creative-git please create an issue following the issue template.

**Comment by Salakar** [comment]
Hey all 👋 

Just published a new "v6" release. Quite a lot changed on this one so I minor versioned it as 0.1.0 for each package (not semver until v6 release).

**Notable Changes**

 - 📦 **Cloud Firestore** initial release for v6
 - 📦 **Cloud Messaging** initial release for v6 (no notifications) [[changelog]](https://github.com/invertase/react-native-firebase/blob/master/CHANGELOG.md#cloud-messaging-messaging)
 - 📦 **ML Kit Vision** initial release for v6 - [[docs]](https://invertase.io/oss/react-native-firebase/v6/ml-vision/reference/module)
 - 📦 **Dynamic Links** initial release for v6
 - Updated to latest Firebase Android SDKs - this means Android X
 - Switched all Firebase Android SDKs to use the [Firebase BoM (v21.1.0)](https://firebase.google.com/support/release-notes/android#firebase_bill_of_materials_firebase_bom_version_2110)

**Other Changes**

 - Lots of new docs (mostly reference docs)
   - App / Core [reference docs](https://invertase.io/oss/react-native-firebase/v6/app/reference) now available (had to rework types on all modules)
   - `firebase.json` [reference docs](https://invertase.io/oss/react-native-firebase/v6/app/reference/firebasejsonconfig) now available 
 - [storage] `Path/Native` statics has been moved to utils now
  - utils will be moved into Core before v6 (currently it's own module)
 - [storage] rename method `getFile` to `writeToFile

See the commit history for other changes.

**Known Issues**

 - Cloud Firestore has two `require cycle` warnings - safe to ignore, will be fixed for next release
 - ML Kit Vision is currently **Android** only - iOS is currently not pushed up as we have a build issue that needs fixing 
 - Dynamic Links changelog out-dated (e.g. it's now `firebase.dynamicLinks()`
   - **Note**: this one is still going through testing so there may be several issues

**Comment by Salakar** [comment]
Another release up, small changes;

 - [STORAGE][NEW] Added support for `StorageReference.list()` & `StorageReference.listAll()`. - @Ehesp 
 - [FIRESTORE][BUGFIX] Fix metro bundler circular require warnings. - @ehesp

Version: `0.1.2`

**Comment by congnguyen91** [comment]
Hi, Please help me where I can set "firebase-bom"?
```
implementation platform("com.google.firebase:firebase-bom:${ReactNative.ext.getVersion("firebase", "bom")}")
```

**Comment by Salakar** [comment]
@congnguyen91 hi, do you mean you need to customise the version or is something erroring saying its missing?

**Comment by congnguyen91** [comment]
@Salakar At the moment I am working with both of AndroidX & Non-AndroidX project. So I want to test or downgrade firebase-version to avoid build error for non-androidx.

**Comment by maxbeech** [comment]
Hi there,

How do we upgrade from v5 please? I can't find info on it anywhere. Ive tried these so far but no luck (I'm using npm not yarn)

`npm install react-native-firebase@6.0.0`
`npm install @react-native-firebase/app@alpha`
`npm install @react-native-firebase/app@6.0.0-alpha`

Thanks a ton!

**Comment by Salakar** [comment]
@congnguyen91 ah, yes you can, add the following to your `app/build.gradle` to customise the BoM version;

```groovy
project.ext {
  set('react-native', [
    versions: [
      firebase: [
        bom: "21.1.0", // change me
      ],
    ],
  ])
}
```

**Comment by Ehesp** [comment]
Version `0.1.3` released:

- [firestore] Fixed an issue with `doc().onSnapshot()` not returning a valid `DocumentSnapshot`. https://github.com/invertase/react-native-firebase/issues/2449
- [database] Fixed an issue with `snapshot` not using the correct reference. https://github.com/invertase/react-native-firebase/issues/2448
- [auth] Fixed an issue with `isBoolean` not being defined when using `verifyPhoneNumber`. https://github.com/invertase/react-native-firebase/issues/2321
- [database][ios] Now correctly cleans up on RN reload. https://github.com/invertase/react-native-firebase/issues/2455
- [database][ios] Podspec now includes correct dependency. https://github.com/invertase/react-native-firebase/issues/2454
- [vision] Vision renamed from `mlVision` to `vision`.
- [vision] iOS integration now complete.
- [language] Natural Language renamed from `mlNaturalLanguage` to `naturalLanguage`.
- [dynamic-links] `onLink` & `getInitialLink` now provide a DynamicLink object. https://github.com/invertase/react-native-firebase/commit/f66ecc9edeca1267d8fb1ee0cb8ef80dab9f1b2c
- [dynamic-links][ios] Fix issue with listeners not being correctly called.
- [firestore] Fixed an issue with settings being called after usage. https://github.com/invertase/react-native-firebase/commit/10a2cbf306923cfb0f8e2b97247b20ea637ea0f8
- [auth] Fixed an issue with `actionCodeSettings` not being correctly applied. https://github.com/invertase/react-native-firebase/issues/2316

**Comment by holgersindbaek** [comment]
@maxbeech Did you figure this out? I need to use v6 because of this issue (https://github.com/invertase/react-native-firebase/issues/2305), but I can't find any guides on how to use v6.

**Comment by Ehesp** [comment]
@holgersindbaek have you looked at https://invertase.io/oss/react-native-firebase/

**Comment by maxbeech** [comment]
@holgersindbaek sadly not, sorry. I just downgraded :(

**Comment by kevindoole** [comment]
Hey folks,

Great work, love the project. I'm considering upgrading because i want to use Firestore's `collectionGroup` which appears to require react-native-firebase 6. It seems unlikely i could upgrade only firestore and continue using the rest of the rnf5 modules my code relies on. Can i pick and choose which modules i upgrade? Or i need to do a wholesale upgrade to 6?

**Comment by mikehardy** [comment]
wholesale upgrade only. I think it's down to just notifications not on v6 though, so it's rapidly becoming possible for most, and soon all people to move to it. (for people completely dependent on admob revenue - that one is active development, so be careful to test things and don't go broke...)

**Comment by Ehesp** [comment]
Version `0.2.3` released (breaking changes):

- [admob] First pass of AdMob has been released which includes AdsConsent, Interstitial and Rewarded Ads. Please see [documentation](https://invertase.io/oss/react-native-firebase/v6/admob) for more info.
  - The `AdsConsent` helper implements Googles own [Consent SDK](https://github.com/googleads/googleads-consent-sdk-android) for Android & iOS. This comes as default with the AdMob module.
  - The AdMob SDKs have been upgraded to the latest versions, which include breaking changes.
  - Ads can now send additional network extras and request only non-personalized ads (for GDPR regulations).
  - `RewardedVideos` is now `RewardedAd` - using the new APIs, rewarded ads can be controlled via the dashboard to display both video and interactive content.
  - Rewarded content can now be pre-fetched and returned when the ad has loaded.
- [fiam][**BREAKING**] The `fiam` module has now been renamed to `inAppMessaging`.
- [config][**BREAKING**] The `config` module has now been renamed to `remoteConfig`.

Testing AdMob using your own AdMob account would be very much appreciated. We will be releasing Banner and Native (pending review) ads in the coming weeks.

**Comment by AshleyColemanBigHealth** [comment]
I have an existing android/ios app that uses the GA analytics. I'm only only need _Analytics_, and maybe _Performance_. From quick look over the project, these pieces look pretty solid in v6 and I'm tempted to skip v5 . Is that overly optimistic?

Currently on react-native 59.1, and using fastlane for CI

thanks

**Comment by mikehardy** [comment]
@AshleyColemanBigHealth might still pay off to go to RN60 and then use RNFB6 ? the effort to move from RNFB5 to RNFB6 later might about the same as RN59->RN60 now, so maybe one work chunk vs 2 total

**Comment by dcolin** [comment]
@Ehesp I'm using this version and `AdsConsentFormOptions ` does not work correctly when an option is set to `false`:

```jsx
{
  withAdFree: false,
  withNonPersonalizedAds: true,
  withPersonalizedAds: false,
}
```

BTW is there a solution to translate the Google ConsentForm?

![google-consentformhtml](https://user-images.githubusercontent.com/2036784/63423550-d133df00-c40c-11e9-850d-60983a67a09a.png)

**Comment by Ehesp** [comment]
@dcolin could you ping me on Discord (Alias) if possible? Would be good to investigate.

**Comment by dcolin** [comment]
@Ehesp In fact, options works on Android devices! But on iOS there is always the three buttons...

**Comment by Ehesp** [comment]
@dcolin ok cheers let me have a look :D

**Comment by Ehesp** [comment]
@dcolin fixed it. Looks like I had my JavaScript hat on when doing the ObjC code... woops! https://github.com/invertase/react-native-firebase/commit/4a8ad14ccc09c1ab5e370fe56d51aaabfe7f757e

**Comment by dcolin** [comment]
@Ehesp Perfect, works fine!

**Comment by cmmartin** [comment]
Anyone else using v6 phone auth? I just upgraded to `"@react-native-firebase/auth": "^0.2.3",` and I am getting the captcha every time I sign in. Normally, I'd assume an issue with my setup, but the same setup is working in v5 without a captcha. I uploaded my APNs Authentication Key to the firebase console, added the URL type and enabled push notifications in XCode.

**Comment by michaelkowalski** [comment]
@mikehardy forgive me if this is obvious but how do we actually install the version 6?

**Comment by alexmobileinteraction** [comment]
Hi!
I'm trying out version 6 and the ml kit with label/text recognition.
The imageLabelerProcessImage function worked great but textRecognizerProcessImage makes the app crash on ios on most images I pass to that function.
Anyone else having similar issues?

**Comment by Salakar** [comment]
@alexmobileinteraction would you mind making a separate issue for that with the log of the crash as well, for reference, this is how we test it and currently not seeing any issues: https://github.com/invertase/react-native-firebase/blob/master/packages/ml-vision/e2e/text.e2e.js

Thanks

**Comment by technoplato** [comment]
> Hey folks,
> 
> Great work, love the project. I'm considering upgrading because i want to use Firestore's `collectionGroup` which appears to require react-native-firebase 6. It seems unlikely i could upgrade only firestore and continue using the rest of the rnf5 modules my code relies on. Can i pick and choose which modules i upgrade? Or i need to do a wholesale upgrade to 6?

You *can* use the vanilla js Firebase package simultaneously with RNF. I tried for the exact same reason as you (for collectionGroups) and all you have to do is manually initialize firebase. Then, once v6 is completely ready, you can remove the firebase js module and upgrade everything to v6.

**Comment by pistou** [comment]
> @pistou there isn't I'm afraid. If you can wait until tomorrow, we'll get this released as it's almost ready to go.

@Ehesp I saw you made some changes on AdMob, adding `BannerAd` according to the commentaries. Unfortunately, even when I *uninstall* then *reinstall*  `@react-native-firebase/admob`, `BannerAd` is not available.

*I feel like* there is an issue with NPM, since the version number didn't change (`v0.3.0`). It may be great to upgrade to `v0.3.1` - assuming that the issue is coming from version number?

As a workaround I downloaded the sources and manually replaced files in `<project>\node_modules\@react-native-firebase\admob\` (tedious) ; it seems to work great though.

**Comment by Salakar** [comment]
@pistou 0.4.0 is now published with this in, please try it out, thanks

**Comment by pistou** [comment]
@Salakar Just tried and it gives an error: 
> You've attempted to require '@react-native-firebase/admob' version '0.4.0', however, the '@react-native-firebase/app' module is of a different version (0.3.0).
> 
> All React Native Firebase modules must be of the same version. Please ensure they match up in your package.json file and re-run yarn/npm install.

I'm not sure why, for me the point of having separate modules was a good way of having them in different advanced levels (*i.e. separated version numbers*)

**Comment by Salakar** [comment]
They all need to be the same version, you'll need to update `app` also to 0.4.0, they are version dependent.

The point of splitting them was separation, yes, but not for versions; with native code & native dependencies in play, we choose to enforce all packages being on the same version number so it's there are no native code conflicts, same as the Firebase iOS SDK does.

**Comment by pistou** [comment]
My bad for the missunderstanding there.
I upgraded every of my `@react-native-firebase` packages and it works perfectly !

Thanks for your work and time.

**Comment by Salakar** [comment]
@pistou no problem 👍 glad it's working, if you have any issues with v6 please let us know via a new issue (tag it with `[v6]` in the name), thank you

**Comment by kaioduarte** [comment]
Has the package been removed? I can't install v6, I'm following the [migration guide](https://invertase.io/oss/react-native-firebase/migrating-to-v6).

![image](https://user-images.githubusercontent.com/11046907/65002234-b872f800-d8ea-11e9-98d7-0fc4b4300e53.png)

**Comment by Ehesp** [comment]
@kaioduarte woops - that was supposed to be `/app`. Just pushed up a fix.

**Comment by kevinvugts** [comment]
@Ehesp 

Hi Elliot,

Could it be possible that crashlytics in v6 is not compatible yet? I followed the docs https://invertase.io/oss/react-native-firebase/v6/crashlytics/quick-start

### and received an error when trying to manually link:

```[!] CocoaPods could not find compatible versions for pod "Fabric":
  In snapshot (Podfile.lock):
    Fabric (= 1.10.2, ~> 1.10.2, ~> 1.6)

  In Podfile:
    ExpoKit/Core (from `http://github.com/expo/expo.git`, tag `ios/2.12.4`) was resolved to 34.0.0, which depends on
      Fabric (~> 1.6)

    RNFBCrashlytics (from `../node_modules/@react-native-firebase/crashlytics/ios`) was resolved to 0.4.3, which depends on
      Fabric (~> 1.9.0)

It seems like you've changed the constraints of dependency `Fabric` inside your development pod `RNFBCrashlytics`.
You should run `pod update Fabric` to apply changes you've made.
```

### pod
pod 'RNFBCrashlytics', :path => '../node_modules/@react-native-firebase/crashlytics/ios'

**Comment by JimTeva** [comment]
In case it can help someone to install "v6.0.0." on **iOS** with a project that is on react-native@0.60.x:

From what I understood, there is not really a version 6 "react-native-firebase@6.x.x". The whole react native firebase library has been splited in different modules, with "@react-native-firebase/app" as the main module to be installed first.
Also depending on the Firebase services that you use, you will need to install the corresponding modules later.

Now, each module/service has its own version, with the main module "@react-native-firebase/app" currently being at version 0.4.3.

**For anyone having issues with the line "@import Firebase;" during iOS install:**
I have added to Xcode > project target > build settings > header search paths and library search paths, and it works fine after that (don't forget to clean build and derivated data first)

Header search paths
"${PODS_ROOT}/Headers/Public/Firebase"
"${PODS_ROOT}/Headers/Public/FirebaseCore"
"${PODS_ROOT}/Headers/Public/FirebaseInstanceID"
"${PODS_ROOT}/Headers/Public/RNFBApp"
${PODS_ROOT}/Firebase/CoreOnly/Sources

Library search paths
"${PODS_CONFIGURATION_BUILD_DIR}/FirebaseCore"
"${PODS_CONFIGURATION_BUILD_DIR}/FirebaseInstanceID"
"${PODS_CONFIGURATION_BUILD_DIR}/RNFBApp"

Source for the paths details: the React Native CLI + Firebase here https://invertase.io/oss/react-native-firebase/quick-start/new-project

Also make sure that the Xcode build phases are empty of the React Native Firebase .a file.

**Comment by Salakar** [comment]
>     ExpoKit/Core (from `http://github.com/expo/expo.git`, tag `ios/2.12.4`) was resolved to 34.0.0, which depends on
>       Fabric (~> 1.6)
> 
>     RNFBCrashlytics (from `../node_modules/@react-native-firebase/crashlytics/ios`) was resolved to 0.4.3, which depends on
>       Fabric (~> 1.9.0)
> 

@kevinvugts, unfortunately, it looks like ExpoKit is bundling its own outdated Fabric version which is conflicting with RNFB, not much we can do here tbh unless you can override the version in Expo somehow to match ours (1.9.0).

**Comment by Salakar** [comment]
@JimTeva correct yes, though all the separate packages are still version tied - they all are published at the same version numbers and will error if non-matching versions are used. Once we officially release we'll graduate all these packages to v6.0.0 and start from there.

I'm not sure why you're having header issues though, we run RN 60.x on our tests app and also the template is RN 60.x - not had any issues yet, very strange

If there's anything lacking on documentation we'd love PRs

**Comment by marcelschulze** [comment]
For when is the notifications module planned to be released? Am I missing something or isn't push notifications one of the most important things? How is anybody able to test v6 yet. Or am I way to opinionated on this one? 

In any way, thank you for your awesome work of course!

**Comment by dcolin** [comment]
I want to use `NativeAd` in my application, is it possible with v6?

BTW the `BannerAd` document page is broken : https://invertase.io/oss/react-native-firebase/v6/admob/reference/bannerad/

**Comment by kevinvugts** [comment]
> @JimTeva correct yes, though all the separate packages are still version tied - they all are published at the same version numbers and will error if non-matching versions are used. Once we officially release we'll graduate all these packages to v6.0.0 and start from there.
> 
> I'm not sure why you're having header issues though, we run RN 60.x on our tests app and also the template is RN 60.x - not had any issues yet, very strange
> 
> If there's anything lacking on documentation we'd love PRs

What would be your advise to do?

**Comment by Ehesp** [comment]
> I want to use `NativeAd` in my application, is it possible with v6?
> 
> BTW the `BannerAd` documentation page is broken : [invertase.io/oss/react-native-firebase/v6/admob/reference/bannerad](https://invertase.io/oss/react-native-firebase/v6/admob/reference/bannerad/)

Native Ads are not supported yet. They're quite tricky, however we do have a plan, it'll just be post-v6.

Also Banner page looks ok, try turning adblocker off?

**Comment by Salakar** [comment]
@kevinvugts I just deployed v0.5.2 to NPM of the pre-release packages; which now allow you to override the Fabric & Crashlytics versions we use internally, update your packages and add this to the top of your `Podfile`

```ruby
$FabricSDKVersion = '1.6.0'
$CrashlyticsSDKVersion = '3.1.0'
```

Documentation [will be added here](https://invertase.io/oss/react-native-firebase/guides/overriding-firebase-versions) alongside the other available overrides.

Note that using early versions of the SDKs means we've not tested they function correctly, so it's at your own risk but should generally be ok.

**Comment by mikehardy** [comment]
> Note that using early versions of the SDKs means we've not tested they function correctly, so it's at your own risk but should generally be ok.

With the important caveat that I think Crashlytics didn't get rid of the UIWebView references receiving "we'll reject your app soon" warnings from the Apple Store, with the community-best-guess indicating they will start rejecting apps for real when iOS 13 is released, which I think is...today https://firebase.google.com/support/releases#september_10_2019

**Comment by Salakar** [comment]
> @mikehardy With the important caveat that I think Crashlytics didn't get rid of the UIWebView references receiving "we'll reject your app soon" warnings from the Apple Store, with the community-best-guess indicating they will start rejecting apps for real when iOS 13 is released, which I think is...today https://firebase.google.com/support/releases#september_10_2019

👏 yes definitely a big caveat.

I've updated the docs to show all the overridable settings now; https://github.com/invertase/react-native-firebase/blob/docs/pages/guides/overriding-native-sdk-versions.md (will be on the site shortly)

**Comment by kaioduarte** [comment]
I just received this warning while using Cloud Storage:

> Unknown option dependency.platforms.android.buildPatch with value ""implementation project(':@react-native-firebase_storage')"" was found

```json
 "dependencies": {
    "@react-native-firebase/app": "^6.0.0-alpha.25",
    "@react-native-firebase/storage": "^6.0.0-alpha.25",
    "react": "16.9.0",
    "react-native": "0.61.0-rc.3"
  }
```

Any thoughts? Maybe related to https://github.com/invertase/react-native-firebase/issues/2194.

**Comment by pistou** [comment]
@kaioduarte According to [this answer](https://github.com/invertase/react-native-firebase/issues/2580#issuecomment-531515845), I believe `^6.0.0-alpha.25` is deprecated and you should use `^0.4.3` which is the latest version at the moment. 

> please change all your @react-native-firebase/* packages to version ^0.4.0 - we moved the v6 alphas to <1.0.0 patch & minor releases so we could partially gate any breaking changes between the alpha behind versions (wasn't so flexible when using tags)

**Comment by kevinvugts** [comment]
> @kevinvugts I just deployed v0.5.2 to NPM of the pre-release packages; which now allow you to override the Fabric & Crashlytics versions we use internally, update your packages and add this to the top of your `Podfile`
> 
> ```ruby
> $FabricSDKVersion = '1.6.0'
> $CrashlyticsSDKVersion = '3.1.0'
> ```
> 
> Documentation [will be added here](https://invertase.io/oss/react-native-firebase/guides/overriding-native-sdk-versions) alongside the other available overrides.
> 
> Note that using early versions of the SDKs means we've not tested they function correctly, so it's at your own risk but should generally be ok.

I am receiving the following error: 

```[!] No podspec found for `RNFBAnalytics` in `../node_modules/@react-native-firebase/analytics/ios```

But the path seems to be correct tho. I also tried to remove node_modules and reinstall package.json. But no result.

**Comment by Ehesp** [comment]
@kevinvugts can you try ensuring the CLI is up to date?

**Comment by kevinvugts** [comment]
> CLI

My Expo CLI?

**Comment by Ehesp** [comment]
Try running `pod install --repo-update`

**Comment by kevinvugts** [comment]
> Try running `pod install --repo-update`

I used that script.

**Comment by Ehesp** [comment]
Please can we see your `Podfile`, looks like auto-linking isn't working for some reason... Did you manually add the pods?

**Comment by kevinvugts** [comment]
> Please can we see your `Podfile`, looks like auto-linking isn't working for some reason... Did you manually add the pods?

I add them manually yes. Here is the pod file:

https://pastebin.com/4z2PGFi3

**Comment by pistou** [comment]
> Right now I am experiencing the following issue:
> 
> ```
> info Opening /Users/kevin/Enorm/xclsv-500/node_modules/@react-native-firebase/app/lib/internal/RNFBNativeEventEmitter.js with atom
> LSOpenURLsWithRole() failed for the application /Applications/Atom.app with error -10810.
> ```

This looks like an error with Atom IDE, maybe [Atom#3093](https://github.com/atom/atom/issues/3093)?

**Comment by radik** [comment]
Is there any progress on Notifications package?
Is there any way how I can help?

**Comment by technoplato** [comment]
@radik See: https://github.com/invertase/react-native-firebase/issues/2566

TLDR Not really a way to help, but progress is being made.

**Comment by jwaldrip** [comment]
How will this be supported with web. The previous version was the same API as the official package from google. How will that work here?

**Comment by Ehesp** [comment]
> How will this be supported with web. The previous version was the same API as the official package from google. How will that work in 6.0.0?

@jwaldrip What do you mean by "supported with web"?

v6 follows the web SDK too where applicable, and for newer modules such as ML Kit we have developed the API into JavaScript based on the native platforms.

**Comment by Salakar** [comment]
Hey all, v6.0.0 is here :tada: see: https://invertase.io/blog/react-native-firebase-v6

**Comment by pistou** [comment]
@Salakar Congrats on the launch (and the shootout at Firebase Summit 💯 ) 🎉

**Comment by r0b0t3d** [comment]
Hi, thanks for your hard work to make v6 available.
I'm working with typescript and I got problem with auto complete
```
import database from '@react-native-firebase/database';

const usersRef = database().ref('users');
```
vscode can not determine the type of `usersRef`.

**Comment by mikehardy** [comment]
@r0b0t3d this does sound like a problem - can you open a fresh issue with the appropriate version details, explaining the same? It will likely not get attention on this one

**Comment by r0b0t3d** [comment]
@mikehardy Thanks. Issue created #2648

**Comment by kevinvugts** [comment]
@Salakar 

I integrated a few of the v 6.0.0 RNFirebase libraries. (perf, app, analytics and crashlytics)

### My pod file looks as following:

```  pod 'RNFBPerf', :path => '../node_modules/@react-native-firebase/perf'
  pod 'RNFBAnalytics', :path => '../node_modules/@react-native-firebase/analytics'
  pod 'RNFBApp', :path => '../node_modules/@react-native-firebase/app'
  pod 'RNFBCrashlytics', :path => '../node_modules/@react-native-firebase/crashlytics'```

However crashlytics does not seem to connect to firebase. I don't receive anything in my console. I have imported crashlytics and performance in my main entry point App.js like so:

```import crashlytics from '@react-native-firebase/crashlytics';
import perf from '@react-native-firebase/perf';
perf().setPerformanceCollectionEnabled(true);
crashlytics().setCrashlyticsCollectionEnabled(true);
```

What am I doing wrong? All libraries work except for crashlytics.

**Comment by maxKimoby** [comment]
Hello guys, great work. Need notifications package though... why not release all at once... :)

**Comment by pistou** [comment]
@maxKimoby According to [the v6 annoucement](https://invertase.io/blog/react-native-firebase-v6)

> We're almost complete migrating the notifications package, please bear with us. As each module is now its own npm package we did not want to hold up the rest of the React Native Firebase modules any longer than necessary because of the notifications package, we're hoping to publish the notifications package in the next 2-3 weeks, stay tuned!

**Comment by mikehardy** [comment]
I know right? I totally want my free stuff faster. And no excuses, sleep is for weenies, but @maxKimoby no PRs in this repo? :thinking: - I'm not being mean though I swear, just pointing out the reality. It's a mountain of work and can always use more contributors.

More useful than that prattle though - I made a tracking issue for that last package here for news #2566

**Comment by maxKimoby** [comment]
Everyone in this repo worked really hard, we love it, we use it, we respect the work and will wait, as we all do for any open source project. I don't think I have the knowledge or experience to contribute to this repo right now, might have in the future! Thank you for your answer. 

At first glance, this version incredibly clearer and simpler to install and use. Great work and sorry @mikehardy if I my message seemed passif-aggressive, was not the case. I had trouble googling the needed information which you provided me.

**Comment by arashkevich25** [comment]
Currently i have error:
`CocoaPods could not find compatible versions for pod "Crashlytics":
  In Podfile:
    Crashlytics (~> 3.13.2)

    RNFBCrashlytics (from '../node_modules/@react-native-firebase/crashlytics)' was resolved to 6.0.0, which depends on
      Crashlytics (~> 3.12.0)

Specs satisfying the 'Crashlytics (~> 3.13.2), Crashlytics (~> 3.12.0)' dependency were found, but they required a higher minimum deployment target.
`

Who know what is wrong ?

**Comment by Ehesp** [comment]
Please create a new issue if you're having issues, following the issue template so we can help.

Locking the thread.
